domains/transfer/TransferAgreementAutomation/Worker/Service/TransferEngineCoordinator.cs

```diff
@@ -0,0 +1,38 @@
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using DataContext.Models;
+using Microsoft.Extensions.Logging;
+
+namespace TransferAgreementAutomation.Worker.Service;
+
+public interface ITransferEngineCoordinator
+{
+    Task TransferCertificates(TransferAgreement transferAgreement, CancellationToken cancellationToken);
+}
+
+public class TransferEngineCoordinator : ITransferEngineCoordinator
+{
+    private readonly IEnumerable<ITransferEngine> _engines;
+    private readonly ILogger<TransferEngineCoordinator> _logger;
+
+    public TransferEngineCoordinator(IEnumerable<ITransferEngine> engines, ILogger<TransferEngineCoordinator> logger)
+    {
+        this._engines = engines;
+        _logger = logger;
+    }
+
+    public async Task TransferCertificates(TransferAgreement transferAgreement, CancellationToken cancellationToken)
+    {
+        foreach (var engine in _engines)
```

Comment:
We need some ordering og engines? Are we allways 100% sure that Transfer Specific amount comes first? If DI for some reason acts strange and not consistent.

Reply:
I added class TransferAgreementProcessOrderComparer and sort transfer agreements in the worker class.